internal/servers/worker/status.go

```diff
@@ -240,22 +245,31 @@ func (w *Worker) cleanupConnections(cancelCtx context.Context, ignoreSessionStat
 			si.status == pbs.SESSIONSTATUS_SESSIONSTATUS_CANCELING,
 			si.status == pbs.SESSIONSTATUS_SESSIONSTATUS_TERMINATED,
 			time.Until(si.lookupSessionResponse.Expiration.AsTime()) < 0:
-			var toClose int
-			for k, v := range si.connInfoMap {
-				if v.closeTime.IsZero() {
-					toClose++
-					v.connCancel()
-					w.logger.Info("terminated connection due to cancellation or expiration", "session_id", si.id, "connection_id", k)
-					closeInfo[k] = si.id
-				}
+			// Cancel connections without regard to individual connection
+			// state.
+			closedIds := w.cancelConnections(si.connInfoMap, true)
+			for _, connId := range closedIds {
+				closeInfo[connId] = si.id
+				w.logClose(si.id, connId)
 			}
+
 			// closeTime is marked by closeConnections iff the
 			// status is returned for that connection as closed. If
 			// the session is no longer valid and all connections
 			// are marked closed, clean up the session.
-			if toClose == 0 {
+			if len(closedIds) == 0 {
 				cleanSessionIds = append(cleanSessionIds, si.id)
 			}
+
+		default:
+			// Cancel connections *with* regard to individual connection
+			// state (ie: only ones that the controller has requested be
+			// terminated).
+			closedIds := w.cancelConnections(si.connInfoMap, false)
+			for _, connId := range closedIds {
+				closeInfo[connId] = si.id
+				w.logClose(si.id, connId)
```

Comment:
Given it's only two places and only a single line for the function, my personal preference would be to just duplicate the logging lines here and below. To me it's a bit less mental energy to just read the line here than have to remember or jump down to the function to see what it's doing.

Reply:
Take it or leave it, just my $0.02.